state/cleanup.go

```diff
@@ -117,6 +117,39 @@ func newCleanupAtOp(when time.Time, kind cleanupKind, prefix string, args ...int
 	}
 }
 
+func (st *State) cancelCleanupOps(kind cleanupKind, prefix string) ([]txn.Op, error) {
+	if prefix == "" {
+		return nil, nil
+	}
+	col, closer := st.db().GetCollection(cleanupsC)
+	defer closer()
+	var docs []cleanupDoc
+	err := col.Find(
+		bson.D{
+			{
+				Name: "prefix", Value: bson.D{
+					{Name: "$regex", Value: fmt.Sprintf("^%s", prefix)},
```

Comment:
This is dangerous and could match other unintended cleanups.
eg if we had apps for "foo" and "foobar" we would include "foobar" when we just wanted "foo".
I think the intent is to remove both apps and units, so we'd need to do that in a way that we don't accidentally remove other things as well.

Reply:
that's right, making the regex more restricted. thanks